p2p/src/error.rs

```diff
@@ -164,6 +164,8 @@ pub enum P2pError {
     DatabaseFailure,
     #[error("Failed to convert data `{0}`")]
     ConversionError(ConversionError),
+    #[error("Noise protocol handshake error")]
+    NoiseHandshakeError,
```

Comment:
I think it would be better to store the original error here. At least in the form of a string for debug purposes.

Reply:
Good idea, thanks. I did not add original error here because `snowstorm::Error` does not implement `PartialEq` and `Eq`, which is needed for `P2pError`. Replaced it with `String` here.